src/map/skill.cpp

```diff
@@ -10703,26 +10690,58 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 		else if (sd != nullptr)
 			clif_skill_fail( *sd, skill_id );
 		break;
-	case HVAN_CHAOTIC:	//[orn]
+	case HVAN_CHAOTIC:
 		{
-			static const int per[5][2]={{20,50},{50,60},{25,75},{60,64},{34,67}};
-			int r = rnd()%100;
-			i = (skill_lv-1)%5;
-			if(r<per[i][0]) //Self
+			// Chance per skill level
+			static const uint8 chance_homunculus[5] = {
```

Comment:
i suggest to declare it in .hpp as constexpr to dont need alloc this list all time valnimirth use skill (he spam it)

Reply:
I don't think we can constexpr this because of the C++ version we're using (14)

About using std::pair, it would force us to hardcode the homun's chance + master's chance as a number, which I don't really like.
Maybe we could use the real chance values in std::pair form and add them in the else if as I originally intended, I don't see the benefit of doing the addition on compiling time